File: scripts/diet_filter.py
### Integrated Proteomics of HOst-MicrobiomE-Diet (IPHOMED)
### Rafael Valdés-Mas (Elinav Lab)

### Filter dietary proteins based on peptide analysis

#!/usr/bin/env python

import re
import sys

# dietary peptides
diet = {}
with open(sys.argv[1], 'r') as file:
    for line in file:
        line = line.strip()
        cols = line.split('\t')
        if cols[0] == "Query":
            continue
        if int(cols[1]) > 0 and float(cols[3]) >= 100:
            diet[cols[4]] = 0

# preparing output file
diet_file = open(sys.argv[3], 'w')

# dietary proteomics output
with open(sys.argv[2], 'r') as file:
    for line in file:
        line = line.strip()
        cols = line.split('\t')
        if cols[0] == "Protein Accession":
            diet_file.write(line + '\n')
        else:
            if cols[0] in diet and int(cols[9]) > 0:
                diet_file.write(line + '\n')

diet_file.close()

File: scripts/diet_filter.R

### Integrated Proteomics of HOst-MicrobiomE-Diet (IPHOMED)
### Rafael Valdés-Mas (Elinav Lab)

### Filter dietary peptides based on blastp alignment
library(dplyr)

## loading files
args = commandArgs(trailingOnly = TRUE)
genus_database <- read.table(args[1], header=TRUE, sep="\t", quote = "", check.names=FALSE, comment.char="") 
genus_database <- subset(genus_database, select=c("Protein", "Genus_TaxID"))

genus_blastout <- read.table(args[2], header=TRUE, sep="\t", quote = "", check.names=FALSE, comment.char="")
genus_blastout$Protein <- gsub("_[0-9]*$", "", genus_blastout$Query)

# combine files
genus_blastout <- merge( genus_blastout, genus_database, by = "Protein", all.x = TRUE)

# filter
count_peptides <- subset(genus_blastout, evalue<0.01) %>% group_by(Query) %>% arrange(evalue) %>% slice_max(with_ties = TRUE, order_by=bitscore) %>% summarize(Count=length(GenusTaxID[Genus_TaxID==GenusTaxID]))
count_peptides <- as.data.frame(count_peptides)
count_peptides <- subset(count_peptides, Count>0)
count_peptides$Protein <- gsub("_[0-9]*$", "", count_peptides$Query)

write.table(count_peptides, args[3], sep="\t", row.names=FALSE, quote=FALSE)